src/pt/sussyscan/src/eu/kanade/tachiyomi/extension/pt/sussyscan/SussyScan.kt

```diff
@@ -1,34 +1,207 @@
 package eu.kanade.tachiyomi.extension.pt.sussyscan
 
-import eu.kanade.tachiyomi.multisrc.madara.Madara
+import android.annotation.SuppressLint
+import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
-import org.jsoup.nodes.Document
-import org.jsoup.nodes.Element
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
 import java.text.SimpleDateFormat
-import java.util.Locale
-
-class SussyScan : Madara(
-    "Sussy Scan",
-    "https://oldi.sussytoons.com",
-    "pt-BR",
-    SimpleDateFormat("MMMM dd, yyyy", Locale("pt", "BR")),
-) {
-    override val client = super.client.newBuilder()
-        .rateLimit(2)
+import java.util.concurrent.TimeUnit
+
+class SussyScan : HttpSource() {
+
+    override val name = "Sussy Scan"
+
+    override val baseUrl = "https://new.sussytoons.site"
+
+    private val apiUrl = "https://api-dev.sussytoons.site"
+
+    override val lang = "pt-BR"
+
+    override val supportsLatest = true
+
+    // Moved from Madara
+    override val versionId = 2
+
+    private val json: Json by injectLazy()
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(1, 2, TimeUnit.SECONDS)
+        .addInterceptor(::imageLocation)
         .build()
 
-    override val useLoadMoreRequest = LoadMoreStrategy.Never
-    override val useNewChapterEndpoint = true
+    override fun headersBuilder() = super.headersBuilder()
+        .set("scan-id", "1")
+
+    // ============================= Popular ==================================
+
+    override fun popularMangaRequest(page: Int) = GET("$apiUrl/obras/top5", headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val dto = response.parseAs<WrapperDto<List<MangaDto>>>()
+        val mangas = dto.results.map(::toSManga)
+        return MangasPage(mangas, dto.hasNextPage())
+    }
+
+    private fun toSManga(dto: MangaDto) = SManga.create().apply {
+        title = dto.name
+        thumbnail_url = dto.thumbnail
+        Jsoup.parseBodyFragment(dto.description).let { description = it.text() }
+        url = "/obras/${dto.id}/${dto.slug}#${dto.scanId}"
```

Comment:
Use `HttpUrlBuilder`

Reply:
I don't understand why I would use HttpBuilder here, since it has a path segment

Reply:
It is usually better to require _less_ domain knowledge to read/understand source code. In this case you would be reducing the cognitive load required to parse it. Simply by the fact that URL parsing requires some overhead, and is often a source of error.

Here it would be something like:
```kt
setUrlWithoutDomain("$baseUrl/obras".toHttpUrl.newBuilder()
    .addPathSegment(dto.id)
    .addPathSegment(dto.slug)
    .fragment(dto.scanId)
    .build()
)

Reply:
`setUrlWithoutDomain` takes a string, so not sure what parsing you mean.

Reply:
I didn't check the exact function signature, but I mean, if any of the arguments (dto.id, dto.slug, dto.scanId) contains characters such as /, or %, or #, it will break. Probably more, but I haven't read through RFC3986/RFC1738 to know if there are other edge cases.

So to make it work with `setUrlWithoutDomain` you would add `.toString()`